vpr/src/device/rr_graph_builder.h

```diff
@@ -235,13 +264,31 @@ class RRGraphBuilder {
     /* Fast look-up for rr nodes */
     RRSpatialLookup node_lookup_;
 
+    /** .. warning:: The Metadata should stay as an independent data structure than rest of the internal data,
+     *  e.g., node_lookup! */
+
     /* Metadata is an extra data on rr-nodes and edges, respectively, that is not used by vpr
```

Comment:
Suggest to move and merge the comment to the API ``rr_node_metadata()``.
Most developers will read APIs first and then internal data (may not even read). So it is better to provide as much information as possible in the APIs.

Reply:
The comments has been added